terrarium/bi_ci/bi_ci/detect_affected_packages.py

```diff
@@ -17,21 +17,21 @@
 class Config:
     root_dir: Path = attr.ib()
     roots: list[str] = attr.ib()
-    changed_paths: list[str] = attr.ib()
+    changed_paths: list[Path] = attr.ib()
 
 
 def collect_affected_packages(
     refs: list[PkgRef],
-    paths: list[str],
+    paths: list[Path],
 ) -> list[PkgRef]:
     # path relative to project root
     seen = set()
-    pkg_by_path = {str(p.partial_parent_path): p for p in refs}
+    pkg_by_path = {p.partial_parent_path: p for p in refs}
 
     # slow, but should work ...
     for p in paths:
         for pkg_path in pkg_by_path:
-            if pkg_path in p:
```

Comment:
this erroneously adds packages that are prefixes of `p` as strings, but not as paths, e.g. if `p` is `path/to/some_big_file` and `pkg_path` is ``path/to/some`